packages/near-swap-find-provider/lib/NearSwapFindProvider.ts

```diff
@@ -134,8 +136,9 @@ export default class NearSwapFindProvider extends NodeProvider implements Partia
     swapParams: SwapParams,
     initiationTxHash: string
   ): Promise<Transaction<near.NearSwapTransaction>> {
-    const initiationTransactionReceipt = await this.getMethod('getTransactionReceipt')(initiationTxHash)
+    validateSwapParams(swapParams)
 
```

Comment:
In this function before returning, also secret needs to be validated like 
`validateSecretAndHash(tx.secret, swapParams.secretHash)`